Add SmartTowerFan to VeSync Integration

easez88 opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->

## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
Adds support for Levoit Classic 42-Inch Smart Tower Fan (LTF-F422S-WUS) to VeSync. This change has already been added upstream. First Pull Request. Completely understand if I missed something.

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [x] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #124123
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [ ] I have followed the [perfect PR recommendations][perfect-pr]
- [ ] The code has been formatted using Ruff (`ruff format homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

easez88 committed:
Add SmartTowerFan to VeSync Integration rebase

zweckj commented on homeassistant/components/vesync/light.py:
```diff
@@ -142,13 +138,13 @@ class VeSyncTunableWhiteLightHA(VeSyncBaseLight, LightEntity):
     _attr_supported_color_modes = {ColorMode.COLOR_TEMP}
 
     @property
-    def color_temp_kelvin(self) -> int | None:
-        """Return the color temperature value in Kelvin."""
```
kelvin is the new and correct unit

easez88 replied:
Reverted Changes. Light.py should be as it was.

zweckj commented on homeassistant/components/vesync/manifest.json:
```diff
@@ -6,5 +6,5 @@
   "iot_class": "cloud_polling",
   "loggers": ["pyvesync"],
-  "requirements": ["pyvesync==2.1.12"]
+  "requirements": ["pyvesync==2.1.13"]
```
bump in prerequisite PR

easez88 replied:
Added  PR for bump #133578

easez88 committed:
Revert changes to light.py

zweckj commented:
since your tests are failing on the dependency bump, converting this to draft until that is merged

easez88 commented:
Looks like they already had a PR for the bump https://github.com/home-assistant/core/pull/131433.

cdnninja commented:
The vesync library has been updated.  This PR can be updated to move it forward.

cdnninja commented:
@zweckj  for some reason it still shows changes requested but I believe that is corrected and ready for review.

zweckj reviewed (changes requested):
@cdnninja yes, it will show that until someone resets the feedback manually, but that's no issue. But CI is failing with ruff issues

easez88 commented:
@zweckj Fixed ruff error

cdnninja commented:
Could you add a test for this device type to push up the test coverage?

easez88 commented:
@cdnninja I'm sorry, I've been trying to figure this out, but I don't know how to add this device to the tests. I'm happy to keep trying if someone could point me in the right direction.

cdnninja commented:
@iprak  Is this something you could help with? I am sure I may be able to fumble my way through the tests but you will be quicker I am sure!

NoRi2909 committed:
Fix typos / grammar in nasweb integration (#134721)

NoRi2909 committed:
Fix spelling of "MAC (address)" and "Gateway" name (#134724)

andrewsayre committed:
Deprecate HEOS sign_in and sign_out actions (#134616)

rpavlik committed:
Add support for "Lumin Smart Light" LD-0003 (#133328)

autinerd committed:
Bump ruff to 0.8.6 (#134745)

autinerd committed:
Bump openwebifpy to 4.3.1 (#134746)

klaasnicolaas committed:
Bump powerfox to v1.1.0 (#134730)

NoRi2909 committed:
Fix swapped letter order in "°F" and "°C" temperature units (#134750)

Fixes the wrong order "F°" and "C°" for the temperature units.

luc-ass committed:
Bump aranet4 to 2.5.0 (#134752)

update aranet4 to 2.5.0 before pull request

DCSBL committed:
Bumb python-homewizard-energy to 7.0.1 (#134753)

NoRi2909 committed:
Fix spelling of "MAC (address)" and "Slide" name in slide_local (#134747)

mib1185 committed:
Register base device entry during coordinator setup in AVM Fritz!Tools integration (#134764)

* register base device entry during coordinator setup

* make mypy happy

NoRi2909 committed:
Use uppercase "Chime" for product name, fix "MAC address" (#134748)

NoRi2909 committed:
Small fixes in the strings file of the Vera integration (#134780)

Just for consistency …

luc-ass committed:
Add ATTR_MODEL to DeviceInfo for Aranet (#134307)

* add ATTR_MODEL to DeviceInfo

* add tests for device context

* Upstream change type.name --> type.model

* fix test to represent model names

TheJulianJES committed:
Bump ZHA to 0.0.45 (#134726)

lanthaler committed:
Add latest Nighthawk WiFi 7 routers to V2 models (#134765)

Click `WiFi Routers` | `Nighthawk WiFi 7 Router` on https://www.netgear.com/support/ to see the list of devices

cdnninja committed:
Remove unneeded vesync device base class (#134499)

* Remove unneeded entity to make code cleaner

* Update light.py

* Update fan.py

* Typing.

* Update homeassistant/components/vesync/common.py

Co-authored-by: Allen Porter <allen.porter@gmail.com>

* Wrap

---------

Co-authored-by: Allen Porter <allen.porter@gmail.com>

NoRi2909 committed:
Use uppercase "ID" in Home Connect strings (#134783)

NoRi2909 committed:
Fix spelling of "ID" in Roku integration (#134779)

* Fix spelling of "ID" and "Ethernet" in Roku integration

Small commit replacing "id" with "ID" and "ethernet" with "Ethernet".

* Revert entity change

Web editor does not support the necessary tests.

RaHehl committed:
Bump uiprotect to version 7.4.1 (#134829)

bdraco committed:
Bump habluetooth to 3.7.0 (#134833)

allenporter committed:
Update roborock to ensure every room has a name, falling back to a placeholder (#134733)

* Update roborock to ensure every room has a name, falling back to a placeholder

* Change Map to Room

NoRi2909 committed:
Fix spelling of "ID", slightly reword action description (#134817)

This commit fixes the spelling of "ID" (uppercase for abbreviations) and slightly changes the action description to use third-person singular.

The latter ensures proper (machine) translations, keeping a descriptive style.

gjohansson-ST committed:
Pass config entry directly to coordinator in System monitor (#134837)

gjohansson-ST committed:
Set single_config_entry in System monitor manifest (#134838)

gjohansson-ST committed:
Add python_script to strict typing (#134822)

mib1185 committed:
Make ChunkAsyncStreamIterator an aiohttp helper (#134843)

make ChunkAsyncStreamIterator a generic aiohttp helper

gjohansson-ST committed:
Add get device capabilities action call for Sensibo (#134596)

* Add get device capabilities action call for Sensibo

* Tests

* Mod

* Fix services

---------

Co-authored-by: J. Nick Koston <nick@koston.org>

gjohansson-ST committed:
Add horizontal swing to Sensibo (#132117)

* Add horizontal swing to Sensibo

* Fixes

* Only load select if already there

* Remove feature check

* Fixes

* Mods

* Last bits

* Mod

* Fixes

* Mods

* Fix test

---------

Co-authored-by: J. Nick Koston <nick@koston.org>

allenporter committed:
Update Roborock config flow message when an account is already configured (#134854)

ludeeus committed:
Log cloud backup upload response status (#134871)

Log the status of the upload response

iprak commented:
> @iprak Is this something you could help with? I am sure I may be able to fumble my way through the tests but you will be quicker I am sure!

Yes I can take a look at the tests.

ludeeus committed:
Log upload BackupAgentError (#134865)

* Log out BackupAgentError

* Update homeassistant/components/backup/manager.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Update homeassistant/components/backup/manager.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Format

---------

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

squishykid committed:
Bump solax to 3.2.3 (#134876)

gjohansson-ST committed:
Raise ImportError in python_script (#134792)

NoRi2909 committed:
Fix missing sentence-casing etc. in several strings (#134775)

tr4nt0r committed:
Bump pynecil to v4.0.1 (#134852)

Djelibeybi committed:
Bump aiolifx-themes to update colors (#134846)

Diegorro98 committed:
Fix how function arguments are passed on actions at Home Connect (#134845)

gjohansson-ST committed:
Remove deprecated binary sensor battery charging from technove (#134844)

NoRi2909 committed:
Replace "id" with "ID" for consistency across HA (#134798)

joostlek commented on homeassistant/components/vesync/strings.json:
```diff
@@ -50,6 +50,7 @@
             "state": {
               "auto": "Auto",
               "sleep": "Sleep",
+              "advanced_sleep": "Advanced Sleep",
```
```suggestion
              "advanced_sleep": "Advanced sleep",
```

NoRi2909 committed:
Fix spelling of "set up", change "id" to uppercase (#134888)

NoRi2909 committed:
Fix a few typos or grammar issues in asus_wrt (#134813)

NoRi2909 committed:
Use correct uppercase for "ID" and sentence-case otherwise (#134815)

Diegorro98 committed:
Iterate over a copy of the list of programs at Home Connect select setup entry (#134684)

klaasnicolaas committed:
Add heat meter to Powerfox integration (#134799)

jb101010-2 committed:
Bump pysuezV2 to 2.0.1 (#134769)

lboue committed:
Matter Battery replacement icon (#134460)

starkillerOG committed:
Add Decorquip virtual motion blinds integration (#134402)

Lash-L committed:
Add extra failure exceptions during roborock setup (#134889)

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

wlcrs committed:
Bump pymodbus version to 3.8.3 (#134809)

NoRi2909 committed:
Use uppercase for "ID" and sentence-case for "name" / "icon" (#134890)

sdb9696 committed:
Bump python-kasa to 0.9.1 (#134893)

Bump tplink python-kasa dependency to 0.9.1

GlennReilly committed:
fix typo "looses" to "loses" in MQTT configuration message (#134894)

starkillerOG committed:
Fix Reolink playback of recodings (#134652)

balloob committed:
Add extra prompt to assist pipeline and conversation (#124743)

* Add extra prompt to assist pipeline and conversation

* extra_prompt -> extra_system_prompt

* Fix rebase

* Fix tests

bramkragten committed:
Update frontend to 20250106.0 (#134905)

tr4nt0r committed:
Fix wrong power limit decimal place in IronOS (#134902)

mib1185 committed:
Bump py-synologydsm-api to 2.6.0 (#134914)

bump py-synologydsm-api to 2.6.0

balloob committed:
Handle discovering user configured Wyoming flow (#134916)

tr4nt0r committed:
Add `bring_api` to loggers in Bring integration (#134897)

Add bring-api to loggers

eclair4151 committed:
Bump pylutron-caseta to 0.23.0 (#134906)

klaasnicolaas committed:
Bump powerfox to v1.2.0 (#134908)

NoRi2909 committed:
Remove excessive newline code, fix "ID", enhance descriptions (#134920)

zweckj committed:
Add device trackers to enabled_by_default fixture (#134446)

arturpragacz committed:
Revert "Remove deprecated supported features warning in LightEntity" (#134927)

gjohansson-ST committed:
Bump holidays to 0.64 (#134922)

gjohansson-ST committed:
Remove deprecated hdr switch from reolink (#134924)

gjohansson-ST committed:
Remove deprecated config entry import from bluesound (#134926)

balloob committed:
Add support for extra_system_prompt to OpenAI (#134931)

bdraco committed:
Migrate to using aiohttp-asyncmdnsresolver for aiohttp resolver (#134830)

chemelli74 committed:
Add device tracker test for Vodafone Station (#134334)

frenck committed:
Revert "Remove deprecated supported features warning in ..." (multiple) (#134933)

NoRi2909 committed:
Fix spelling of "ID", slightly reword action descriptions (#134778)

RaHehl committed:
UnifiProtect Refactor light control methods to use new API (#134625)

EliSchleifer committed:
add proxy view for unifiprotect to grab snapshot at specific time (#133546)

Co-authored-by: J. Nick Koston <nick@koston.org>

davidrapan committed:
Update Shelly integration: Remove double "Error fetching ..." from error messages (#134950)

refactor: Remove double "Error fetching" from error messages

kelyaenn committed:
Bump renault-api to 0.2.9 (#134858)

* chore: Bump Renault api version

* Update requirements_all.txt

* Update requirements_test_all.txt

bdraco committed:
Bump zeroconf to 0.137.2 (#134942)

NoRi2909 committed:
Change "id" to uppercase for consistency (#134971)

NoRi2909 committed:
Use sentence case, capitalize "IP Secure" and "ID" (#134966)

ludeeus committed:
Increase cloud backup download timeout (#134961)

Increese download timeout

epenet committed:
Simplify onewire config-flow (#134952)

epenet committed:
Set parallel-updates and scan-interval explicitly in onewire (#134953)

allenporter committed:
Update roborock tests to only load the platform under test (#134694)

andrewsayre committed:
Update HEOS Quality Scale docs-related items (#134466)

Update docs items

emontnemery committed:
Fix DSMR migration (#134990)

allenporter committed:
Simplify roborock coordinator (#134700)

* Update roborock coordinator to require maps on startup

* Fix indent in merge

epenet committed:
Add reconfigure to onewire (#134996)

* Add reconfigure to onewire

* Adjust _async_abort_entries_match

emontnemery committed:
Validate device id when adding or updating entity registry entry (#134982)

epenet committed:
Set PARALLEL_UPDATES to 1 in onewire (#135006)

chemelli74 committed:
Use standard "entity_registry_enabled_by_default" fixture (#134962)

Lash-L committed:
Add total cleaning count sensor to Roborock (#135015)

dieselrabbit committed:
Add Controller state sensor to screenlogic (#133827)

epenet committed:
Move OneWireConfigEntry type definition (#135004)

iMicknl committed:
Improve logic for event polling duration in Overkiz (#133617)

allenporter committed:
Bump voluptuous openapi to 0.0.6 (#134998)

iMicknl committed:
Catch errors in automation (instead of raise unexpected error) in Overkiz (#135026)

Catch errors in automation (instead of raise unexpected error)

sdb9696 committed:
Add data description translations to all tplink config flow steps (#135022)

NoRi2909 committed:
Improve Huawei LTE suspend integration service description  (#135021)

Bre77 committed:
Bump pyaussiebb to 0.1.5 (#134943)

Bump

dontinelli committed:
Add new sensors to fyta (#135032)

dgomes committed:
Add weather warning sensor to IPMA (#134054)

Co-authored-by: Abílio Costa <abmantis@users.noreply.github.com>

ludeeus committed:
Implement upload retry logic in CloudBackupAgent (#135062)

* Implement upload retry logic in CloudBackupAgent

* Update backup.py

Co-authored-by: Erik Montnemery <erik@montnemery.com>

* nit

---------

Co-authored-by: Erik Montnemery <erik@montnemery.com>

bdraco committed:
Bump dbus-fast to 2.28.0 (#135049)

changelog: https://github.com/Bluetooth-Devices/dbus-fast/compare/v2.24.3...v2.28.0

miaucl committed:
Bump cookidoo-api to 0.12.2 (#135045)

fix cookidoo .co.uk countries and group api endpoint

Thomas55555 committed:
Bump aioautomower to 2025.1.0 (#135039)

starkillerOG committed:
Fix channel retrieval for Reolink DUO V1 connected to a NVR (#135035)

fix channel retrieval for DUO V1 connected to a NVR

gjohansson-ST committed:
Fix climate react type (#135030)

epenet committed:
Improve type hints in onewire tests (#134993)

farmio committed:
Delete KNX config storage when removing the integration (#135071)

andrewsayre committed:
Update pyheos to 0.9.0 (#134947)

Bump pyheos

puddly committed:
Fix ZHA "referencing a non existing `via_device`" warning (#135008)

starkillerOG committed:
Add Reolink unexpected error translation (#134807)

thecode committed:
Cleanup LG webOS TV name (#135028)

emontnemery committed:
Fix DSMR migration (#135068)

austinmroczek committed:
TotalConnect improved config flow and test before setup (#133852)

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

epenet committed:
Remove rounding from onewire sensors (#135095)

epenet committed:
Add quality_scale file to onewire (#134951)

dontinelli committed:
Add binary sensors to fyta (#134900)

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

david1024pl committed:
Add satel integra switches and alarm control panels unique_id (#129636)

farkasdi committed:
Update addition logger string in fan.py (#135098)

elmurato committed:
Add quality_scale.yaml to Minecraft Server (#132551)

sdb9696 committed:
Add PARALLEL_UPDATES constant to tplink integration platforms (#135083)

sdb9696 committed:
Raise HomeAssistantError from tplink light effect service (#135081)

jb101010-2 committed:
Bump pysuezV2 to 2.0.3 (#135080)

epenet committed:
Refactor patching in onewire tests (#135070)

epenet committed:
Split simple and recovery in onewire config-flow user tests (#135102)

gjohansson-ST committed:
Fix hvac_modes never empty in Sensibo (#135029)

epenet committed:
Improve onewire options flow tests (#135109)

bors-ltd committed:
Improve holidays config form and naming (#133663)

The holidays library is improving over time, let's make use of their
data for a more user-friendly experience.

Co-authored-by: G Johansson <goran.johansson@shiftit.se>

catsmanac committed:
Update enphase_envoy test_init to use str for unique_id and test for loaded config entry (#133810)

sdb9696 committed:
Add quality scale file to tplink integration (#135017)

catsmanac committed:
Remove enphase_envoy config flow tests that make no sense (#133833)

LaStrada committed:
Add suggested precision for Airthings BLE integration (#134985)

Add suggested precision

jpelgrom committed:
Add LetPot integration (#134925)

thecode committed:
Set PARALLEL_UPDATES in LG webOS TV (#135135)

LouisChrist committed:
Add DataUpdateCoordinator to bluesound integration (#135125)

jpelgrom committed:
Use is in FlowResultType enum comparison in integration scaffold tests (#135133)

thecode committed:
Impove LG webOS TV tests quality (#135130)

* Impove LG webOS TV tests quality

* Review comments

eclair4151 committed:
Add support for Lutron Wood Tilt Blinds (#135057)

Quentame committed:
Fix Météo-France setup in non French cities (because of failed next rain sensor) (#134782)

gjohansson-ST committed:
Render select entity unavailable when active feature is missing in Sensibo (#135031)

gjohansson-ST committed:
Change minimum SQLite version to 3.40.1 (#135042)

Co-authored-by: J. Nick Koston <nick@koston.org>

iprak commented:
@easez88 

SmartTowerFan is defined as following in https://github.com/webdjoe/pyvesync/blob/master/src/pyvesync/vesyncfan.py

```
   'SmartTowerFan': {
        'module': 'VeSyncTowerFan',
        'models': ['LTF-F422S-KEU', 'LTF-F422S-WUSR', 'LTF-F422_WJP', 'LTF-F422S-WUS'],
        'modes': ['normal', 'auto', 'advancedSleep', 'turbo', 'off'],
        'set_mode_method': 'setTowerFanMode',
        'features': ['fan_speed'],
        'levels': list(range(1, 13))
    }
	
class VeSyncTowerFan(VeSyncAirBaseV2):
class VeSyncAirBaseV2(VeSyncAirBypass):
```

The class VeSyncTowerFan represents SmartTowerFan which pulls data from '/cloud/v2/deviceManaged/bypassV2' via POST and then parses the result through [build_purifier_dict](https://github.com/webdjoe/pyvesync/blob/736c29256a5a8b562ddb1b16ac20e418b3e27f29/src/pyvesync/vesyncfan.py#L375) and [build_config_dict](https://github.com/webdjoe/pyvesync/blob/736c29256a5a8b562ddb1b16ac20e418b3e27f29/src/pyvesync/vesyncfan.py#L375)

Why fan uses build_purifier_dict is a question for someone else but let us go with what it is.

To create correct test data you will have to run in debug mode and see what the POST returns but one can stuff in some fake data based on code analysis. And that is what I have done in https://github.com/iprak/core-fork/tree/SmartTowerFan-tests which is based on your branch.

It would be nice if you can put in some more relevant data into SmartTowerFan-detail.json.

The snapshot updates were generated by running `pytest tests/components/vesync --snapshot-update`. Your branch is slightly old so you might have to fix some failures.

gjohansson-ST committed:
Make devices dynamic in Sensibo (#134935)

ZephireNZ committed:
Fix Flick Electric Pricing (#135154)

Anonym-tsk committed:
Deprecate StarLine engine switch attributes (#133958)

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

bdraco committed:
Bump zeroconf to 0.138.1 (#135148)

emontnemery committed:
Add jitter to backup start time to avoid thundering herd (#135065)

epenet committed:
Deprecate raw_value attribute in onewire entity (#135171)

* Drop raw_value attribute in onewire entity

* Deprecate only

jpbede committed:
Mark docs-installation-parameters and docs-removal-instructions for inexogy as done (#135126)

crevetor committed:
Add outside temperature sensor to fujitsu_fglair (#130717)

epenet committed:
Move OneWire PLATFORM constant back to init (#135172)

sdb9696 committed:
Add and cleanup tplink translations (#135120)

miaucl committed:
Add uuid as unique_id to config entries for Cookidoo (#134831)

epenet committed:
Use snapshot_platform helper in onewire tests (#135176)

* Use snapshot_platform helper in onewire tests

* Snapshot device registry

epenet committed:
Set PARALLEL_UPDATES to 0 in onewire (#135178)

beginner2047 committed:
Add yue language support to Google Translate TTS (#134480)

sdb9696 committed:
Fix stale docstrings in tplink integration (#135183)

epenet committed:
Update scaffold to use internal _PLATFORM constant (#135177)

epenet committed:
Add select platform to onewire (#135181)

* Add select platform to onewire

* Add tests

* Apply suggestions from code review

joostlek committed:
Add event entities to Overseerr (#134975)

NoRi2909 committed:
Update title and description for setup dialog of thethingsnetwork (#134954)

kereyroper committed:
add support for dimming/brightening X10 lamps (#130196)

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

catsmanac committed:
No need to set unique_id in enphase_envoy reauth step (#133615)

Co-authored-by: epenet <6771947+epenet@users.noreply.github.com>

alengwenus committed:
Update module properties on module scan for LCN (#135018)

NoRi2909 committed:
Use friendly names in add_to_playlist action, fix "ID" (#134978)

epenet committed:
Avoid unnecessary executor calls in onewire (#135187)

jbouwh committed:
Allow to process kelvin as color_temp for mqtt basic light (#133953)

allenporter committed:
Allow unregistering LLM APIs (#135162)

KapJI committed:
Make generated files appear as generated (#134991)

sdb9696 committed:
Use typed config entry in tplink coordinator (#135182)

epenet committed:
Refactor onewire hub (#135186)

* Improve type hints in onewire hub

* More cleanups

* Improve

* Get host/port from entry data

* Use DeviceInfo object

thecode committed:
Fix LG webOS TV init test coverage (#135194)

jbouwh committed:
Allow to process kelvin as color_temp for mqtt json light (#133955)

Bre77 committed:
Add streaming sensors to Teslemetry (#132783)

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

bdraco committed:
Bump zeroconf to 0.139.0 (#135213)

epenet committed:
Use remove-prefix/suffix introduced in Python 3.9 (#135206)

Use removeprefix/removesuffix

epenet committed:
Add exception-translations rule to quality_scale pytest validation (#131914)

* Add exception-translations rule to quality_scale pytest validation

* Adjust

* Return empty dict if file is missing

* Fix

* Improve typing

* Address comments

* Update tests/components/conftest.py

* Update tests/components/conftest.py

* Update tests/components/conftest.py

---------

Co-authored-by: Robert Resch <robert@resch.dev>

chemelli74 committed:
Add support for Shelly BLU TRV (#128439)

* feat: add support for Shelly BLU TRV

* chore: apply some fixes

* make BLUTRV a separate device

* apply review comment

* review comments and small optimization

* add HVACMode.OFF

* a couple of fixes

* 2 more fixes

* better approach

* cleanup

* small optimization

* remove cooling as not supported by firmware

* tweaks

* humidity and entity name

* fix naming

* allign async_set_hvac_mode

* align settings

* restore temp

* fix

* remove OFF

* cleanup

* hvac_mode

* add tests

* typo

* more tests

* bump aioshelly

bdraco committed:
Bump anyio to 4.8.0 (#135224)

bramkragten committed:
Update frontend to 20250109.0 (#135235)

chemelli74 committed:
Refactor Vodafone Station tests (#134956)

bdraco committed:
Bump fnv-hash-fast to 1.1.0 (#135237)

peteS-UK committed:
Expand Squeezebox auth test for config_flow to finish on create_entry (#133612)

Expand auth test to create_entry

bdraco committed:
Remove per engine max bind vars (#135153)

NoRi2909 committed:
Make description of alarm_arm_vacation consistent (#135257)

Small fix to also use "Arms …" in the description of the alarm_arm_vacation action, making it consistent with the other two alarm_arm_… actions.

epenet committed:
Improve formatting in core files (#135256)

* Adjust core files formatting

* Adjust translations script

epenet committed:
Split long strings in components (#135263)

epenet committed:
Combine short byte strings in xiaomi_ble tests (#135268)

thecode committed:
Fix LG webOS TV media player test coverage (#135225)

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

epenet committed:
Improve formatting in component test files (#135267)

Improve formatting in test files

NoRi2909 committed:
Change 'entity_id' to UI-friendly 'Entity ID', fix spelling of "setpoint" (#135234)

In addition this makes the description of the first action consistent, using third-person singular like the other two and adhering to the HA standard.

epenet committed:
Combine short strings in components (#135265)

epenet committed:
Split long strings in stream hls tests (#135271)

epenet committed:
Fix missing comma in ollama MODEL_NAMES (#135262)

cereal2nd committed:
Velbus light platform code cleanup (#134482)

epenet committed:
Fix incorrect test in test_core_config (#135260)

epenet committed:
Add model_id and serial_number to onewire device info (#135279)

jbouwh committed:
Allow to process kelvin as color_temp for mqtt template light (#133957)

epenet committed:
Improve formatting in component files (#135261)

* Improve formatting in component files

* Apply suggestions from code review

epenet committed:
Discover new devices at runtime in onewire (#135199)

cdnninja committed:
Simplify vesync init loading (#135052)

thecode committed:
Move LG webOS TV actions to entitiy services (#135285)

DCSBL committed:
Adjust HomeWizard to use updated python-homewizard-energy library (#135046)

NoRi2909 committed:
Smaller grammar fixes, replace 'entity_id' with UI-friendly 'ID' (#135236)

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

crevetor committed:
Mark FGLAir entities unavailable if they are reporting to be offline (#135202)

dotvav committed:
Bump pypalazzetti to 0.1.16 (#135269)

NoRi2909 committed:
Replace 'entity_id' with UI-friendly, localizable 'entity ID' (#135232)

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

tr4nt0r committed:
Add USB-PD Mode select entity to IronOS integration (#134901)

Add USB-PD Mode select entity

NoRi2909 committed:
Grammar and consistency fixes in hdmi_cec strings (#135292)

bieniu committed:
Bump aioshelly to version 12.3.1 (#135299)

thecode committed:
Use runtime_data in LG webOS TV (#135301)

NoRi2909 committed:
Fix typos and inconsistent spelling of "tedee" brand name (#135305)

- Change "Setup your tedee locks" to "Set up …"
- Remove two excessive commas
- Change one occurrence of "Tedee" to "tedee".

NoRi2909 committed:
Fix typos, replace duplicated strings with references (#135303)

Quentame committed:
Bump Freebox to 1.2.2 (#135313)

edenhaus committed:
Don't store uv's lockfile in hassfest image (#135214)

tr4nt0r committed:
Fix Habitica gems/hourglass sensors (#135323)

DCSBL committed:
Use LOGGER from homewizard.const instead per-file loggers (#135320)

bdraco committed:
Bump habluetooth to 3.8.0 (#135322)

changelog: https://github.com/Bluetooth-Devices/habluetooth/compare/v3.7.0...v3.8.0

balloob committed:
Extract resolve announcement media ID for AssistSatelliteEntity (#134917)

bdraco committed:
Add bluetooth subscribe_advertisements WebSocket API (#134291)

NoRi2909 committed:
Fix spelling of "Log in", fix "outdated student" (#135348)

NoRi2909 committed:
Small fixes in setup flow strings, correct sentence-case (#135349)

silamon committed:
Add config flow to NMBS (#121548)

Co-authored-by: Joostlek <joostlek@outlook.com>

joostlek committed:
Fix backup formatting (#135350)

dependabot[bot] committed:
Bump actions/upload-artifact from 4.5.0 to 4.6.0 (#135255)

dependabot[bot] committed:
Bump docker/build-push-action from 6.10.0 to 6.11.0 (#135254)

silamon committed:
Remove code owner for nmbs (#135357)

NoRi2909 committed:
Fix spelling of "Log in …" and "API key" in LOQED integration (#135347)

erwindouna committed:
Fix Tado config flow (#135353)

silamon committed:
Add more typings to nmbs sensor (#135359)

dotvav committed:
Palazzetti Quality Scale update after doc improvement (#135277)

Bre77 committed:
Bump Teslemetry Stream (#135344)

bump

lanthaler committed:
Enable slowly-changing, important diagnostics for connected devices by default (#134776)

thecode committed:
Add reconfigure to LG webOS TV (#135360)

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

jterrace committed:
Add @jterrace to onvif integration owners (#135398)

YogevBokobza committed:
Switcher runner child lock support (#133270)

* Switcher runner child lock support

* fix based on requested changes

* Update homeassistant/components/switcher_kis/switch.py

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* Fix

---------

Co-authored-by: Shay Levy <levyshay1@gmail.com>
Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

NoRi2909 committed:
Make all three numbered lists consistent, using 1. 1. 1. for the syntax (#135400)

Make all three numbered lists use 1. 1. 1. for the syntax

Currently only two of the setup descriptions of the Nest integration use automatic syntax for a numbered list.

This commit makes the third one consistent, using 1. 1. 1. as well.

This helps translators in Lokalise understand that this is the expected format for all numbered lists in Home Assistant.

jterrace committed:
Add additional Tapo ONVIF Person/Vehicle/Line/Tamper/Intrusion events (#135399)

NoRi2909 committed:
Reword action descriptions for better translations in Teslemetry (#135370)

Slightly reword action descriptions for better translations

Currently only one of the action descriptions in the Teslemetry integration uses the descriptive form of third person plural.

This commit changes the remaining descriptions to adopt the same language and changes "the" to "a" as the actual action target is defined below that in the UI.

tr4nt0r committed:
Add sensors for inventory items to Habitica (#135331)

Add sensors for inventory items

joostlek committed:
Actually use translated entity names in Lametric (#135381)

andrewsayre committed:
Bump pyheos to 1.0.0 (#135415)

tronikos committed:
Fix config entries typo s/entruis/entries/ (#135431)

Fix typo s/entruis/entries/

NoRi2909 committed:
Fix descriptions of send_message action of Bring! integration (#135446)

* Make "Urgent message" selector consistent, use "Bring!" as name

- Replace one occurrence of "bring" with the brand name "Bring!"
- Change description of action to third-person singular for consistency in Home Assistant
- Make all occurrences of the selector "Urgent message" consistent (in sentence case) so they all get consistent translations, too
- Change one related error message to refer to the UI name of the required "Article" field

* Changed ` to '  to avoid Regex problems

* Reverted change to notify_missing_argument_item

Reverted to avoid failing test

* Reverted change to "bring"

* Add "is" to description of "Article"

Co-authored-by: Jan Bouwhuis <jbouwh@users.noreply.github.com>

---------

Co-authored-by: Jan Bouwhuis <jbouwh@users.noreply.github.com>

WaterInTheLake committed:
Fix translation string: numbering in list (#135441)

easez88 committed:
Added Tests for SmartTowerFan

cdnninja commented:
I would recommend closing this PR and creating a new one based on a new branch.  Something went wrong on the last merge.

cdnninja commented:
@homeassistant close

cdnninja commented:
@home-assistant close
